src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolWorkQueue.cs

```diff
@@ -1252,9 +1252,14 @@ private static void DispatchWorkItem(object workItem, Thread currentThread)
             else
             {
                 Debug.Assert(workItem is IThreadPoolWorkItem);
+                IThreadPoolWorkItem tpWorkItem;
+                unsafe
```

Comment:
Please add `// FIXME: parent should be unsafe and specify input must be Task or IThreadPoolWorkItem`